docs/developers/content-creation/creating-a-weapon.md

```diff
@@ -1,3 +1,120 @@
 # Creating a Weapon
 
-Some Text
+When starting content creation for gmod, creating a basic weapon using the SWEP structure is one of the easiest things you can do.
+For this guide I'll be using the Visual Studio Code with the glualint extension as referenced in the "Creating an addon" section of these documents.
+
+## File Format
+
+Firstly you need to make sure that you're using the correct folder structure.
+Your addon folder must be in this format:
+
+* `weapon_ttt_<weapon name>\gamemodes\terrortown\entities\weapons\weapon_ttt_\<weapon name>.lua`
+
+Your .vmt and .vtf icon files (what gets shown in the buy menu) will need to be stored in this format:
+
+* `\weapon_ttt_<weapon name>\materials\vgui\ttt\<weapon name>_icon.vtf`
+* `\weapon_ttt_<weapon name>\materials\vgui\ttt\<weapon name>_icon.vmt`
+
+See the [Creating An Addon](https://docs.ttt2.neoxult.de/developers/basics/creating-an-addon/) page for more info on folder structure
+
+## Important 'Structural' features
+
+### If SERVER Block
+
+```lua
+if SERVER then
+   -- Shares the specified file with the client (if blank shares current file)
+   AddCSLuaFile()
+
+   -- Shares the buy menu icon with the client
+   resource.AddFile("materials/vgui/ttt/blue_template_icon.vmt")
+end
+```
+It's very important to create a globally unique name for this because the Gmod
+client will look for the given path locally to display it. So if there are two
+identical icon names the client may display the incorrect one
+
+### If CLIENT Block
+
+```lua
+if CLIENT then
+   -- Client viewpoint vals
+   SWEP.PrintName          = "Template Weapon"
+   SWEP.Slot               = 6 -- (See SWEP.Kind below)
+   SWEP.ViewModelFOV       = 70 -- How much of the first person viewmodel is seen
+   SWEP.ViewModelFlip      = true -- If true, the viewmodel is on the left
+   SWEP.DrawCrossHair      = false -- Does the client draw its own crosshair over the default ones
+
+   -- Client menu data
+   SWEP.Icon = "vgui/ttt/blue_template_icon.vtf" -- The filename of the icon shown in the buymenu
+   -- The information shown in the buy menu
+   SWEP.EquipMenuData      =
+   {
+      type = "Weapon",
+      desc = "This is a template weapon"
```

Comment:
I'd reccomend showing how to use translations here. See: 

https://github.com/TTT-2/ttt2-wep_doorlocker/blob/master/gamemodes/terrortown/entities/weapons/weapon_ttt_doorlocker.lua#L21-L25

Reply:
Ah, this is new to me too - I'm afraid I'll probably finish that off tomorrow

Reply:
No hurry. I just added a new article about language support over here: https://docs.ttt2.neoxult.de/developers/content-creation/language-support/

Reply:
ta

Reply:
btw on the docs page you say that each language identifier must be unique, but on your doorlocker they're just the language names. Is this a mistake or have I got language identifier mixed up with the lua file?

Reply:
could you explain what you mean by "not unique"?

Reply:
Oh I've got mixed up then, because i thought you meant this was the language identifier
![image](https://user-images.githubusercontent.com/67487108/86126450-43b62680-bad6-11ea-8e25-d81b7ced9cfe.png) from the lang support page

Reply:
Ah no, the language itself shouldn't be unique, maybe I have to specify this

Reply:
ah aight thank you